theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -267,24 +279,21 @@ def setup_sources_dependency(project, os, version, repoowner, pull_request = fal
     def package_dir
 
     dir(build_dir) {
-        sh """
-            BUILD_TYPE=gem
-            # Import variables from the project, allowing it to override behaviour
-            if [ -e ../${project}/build_vars.sh ]; then
-              . ../${project}/build_vars.sh
-            fi
+        def build_vars = read_build_vars("../${project}")
+        def build_type = build_vars.containsKey('BUILD_TYPE') ? build_vars['BUILD_TYPE'] : 'gem'
 
-            if [ "\$BUILD_TYPE" = "python" ]; then
+        sh """
+            if [ "${build_type}" = "python" ]; then
```

Comment:
given `build_type` is now a Groovy-level variable, should the `if/else` also become Groovy and the `sh` calls move *inside* the `if/else`? IMHO much saner to read, but also scope-creep somehow.